fix(rds): database proxies use ids as their resource names directly (under feature flag)

ymhiroki opened the pull request:
fixes #18578


----

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Construct Runtime Dependencies:

* [ ] This PR adds new construct runtime dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-construct-runtime-dependencies)

### New Features

* [X] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [X] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 26bd3e84db3a1971f6b0842928c11740cfb2911b
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c4f5490c-27fa-4812-b2a5-e5cb76bf4898%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation reviewed (changes requested):
The pull request linter has failed. See the aws-cdk-automation comment below for failure reasons. If you believe this pull request should receive an exemption, please comment and provide a justification.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Fixes must contain a change to an integration test file and the resulting snapshot.
	❌ The title of this pull request does not follow the Conventional Commits format, see https://www.conventionalcommits.org/.

PRs must pass status checks before we can provide a meaningful review.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ The title of this pull request does not follow the Conventional Commits format, see https://www.conventionalcommits.org/.

PRs must pass status checks before we can provide a meaningful review.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 2db23379fa56725128c1f905c96ec53ce144eab8
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=53308383-5b0d-4d37-b78e-2737920f3885%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Fixes must contain a change to an integration test file and the resulting snapshot.
	❌ The title of this pull request does not follow the Conventional Commits format, see https://www.conventionalcommits.org/.

PRs must pass status checks before we can provide a meaningful review.

ymhiroki commented:
I added the testcase in [integ.proxy.ts](https://github.com/aws/aws-cdk/pull/23703/files#diff-786cfbf5519192ee28c635fbb1ad55a22e3352714fc4beab903ca0fea7798a2d) to pass the CI, however, it fails when I run `yarn integ-runner --update-onfailed` with the error below again. (`dbProxy3B89EAF2` seems to be updated even though it doesn't change at all.)

How can I pass the CI or make the snapshot successfully?

```
yarn integ --update-on-failed test/integ.proxy.js
yarn run v1.22.18
$ integ-runner --update-on-failed test/integ.proxy.js

Verifying integration test snapshots...

  CHANGED    integ.proxy 10.971s
      IAM Statement Changes
┌───┬──────────────────────────────────────┬────────┬───────────────────────────────┬─────────────────────────────┬───────────┐
│   │ Resource                             │ Effect │ Action                        │ Principal                   │ Condition │
├───┼──────────────────────────────────────┼────────┼───────────────────────────────┼─────────────────────────────┼───────────┤
│ + │ ${ProxyIAMRole2FE8AB0F.Arn}          │ Allow  │ sts:AssumeRole                │ Service:rds.amazonaws.com   │           │
├───┼──────────────────────────────────────┼────────┼───────────────────────────────┼─────────────────────────────┼───────────┤
│ + │ ${dbClusterSecretAttachmentAB67A752} │ Allow  │ secretsmanager:DescribeSecret │ AWS:${ProxyIAMRole2FE8AB0F} │           │
│   │                                      │        │ secretsmanager:GetSecretValue │                             │           │
└───┴──────────────────────────────────────┴────────┴───────────────────────────────┴─────────────────────────────┴───────────┘
Security Group Changes
┌───┬────────────────────────────────────────────┬─────┬────────────────────────────────────────┬────────────────────────────────────────────┐
│   │ Group                                      │ Dir │ Protocol                               │ Peer                                       │
├───┼────────────────────────────────────────────┼─────┼────────────────────────────────────────┼────────────────────────────────────────────┤
│ + │ ${ProxyProxySecurityGroupC42FC3CE.GroupId} │ Out │ Everything                             │ Everyone (IPv4)                            │
├───┼────────────────────────────────────────────┼─────┼────────────────────────────────────────┼────────────────────────────────────────────┤
│ + │ ${dbClusterSecurityGroupCAA1A91F.GroupId}  │ In  │ TCP ${dbClusterE86E47AE.Endpoint.Port} │ ${ProxyProxySecurityGroupC42FC3CE.GroupId} │
│ + │ ${dbClusterSecurityGroupCAA1A91F.GroupId}  │ Out │ Everything                             │ Everyone (IPv4)                            │
└───┴────────────────────────────────────────────┴─────┴────────────────────────────────────────┴────────────────────────────────────────────┘
(NOTE: There may be security-related changes not in this list. See https://github.com/aws/aws-cdk/issues/1299)

Resources
[+] AWS::RDS::DBSubnetGroup dbClusterSubnets03B9B0E1
[+] AWS::EC2::SecurityGroup dbClusterSecurityGroupCAA1A91F
[+] AWS::EC2::SecurityGroupIngress dbClusterSecurityGroupfromawscdkrdsproxyProxyProxySecurityGroup9F179E6FIndirectPortED421002
[+] AWS::SecretsManager::Secret dbClusterSecretCEA6D7B6
[+] AWS::SecretsManager::SecretTargetAttachment dbClusterSecretAttachmentAB67A752
[+] AWS::RDS::DBCluster dbClusterE86E47AE
[+] AWS::RDS::DBInstance dbClusterInstance1BCE092AC
[+] AWS::RDS::DBInstance dbClusterInstance20BA1ECD9
[+] AWS::IAM::Role ProxyIAMRole2FE8AB0F
[+] AWS::IAM::Policy ProxyIAMRoleDefaultPolicy59EB0117
[+] AWS::EC2::SecurityGroup ProxyProxySecurityGroupC42FC3CE
[+] AWS::RDS::DBProxy ProxyCB0DFB71
[+] AWS::RDS::DBProxyTargetGroup ProxyProxyTargetGroupB462B5C5



Snapshot Results:

Tests:    1 failed, 1 total

Running integration tests for failed tests...

Running in parallel across regions: us-east-1, us-east-2, us-west-2
Running test /Users/yhirokiy/works/ymhiroki/aws-cdk/packages/@aws-cdk/aws-rds/test/integ.proxy.js in us-east-1

✨  Synthesis time: 1.63s

aws-cdk-rds-proxy: building assets...

[0%] start: Building eade7d2d48419acaec1228343071922ec168779f7a6dd46405c9605c5652b43e:current_account-current_region
[100%] success: Built eade7d2d48419acaec1228343071922ec168779f7a6dd46405c9605c5652b43e:current_account-current_region

aws-cdk-rds-proxy: assets built

aws-cdk-rds-proxy: deploying... [1/1]
[0%] start: Publishing eade7d2d48419acaec1228343071922ec168779f7a6dd46405c9605c5652b43e:current_account-current_region
[100%] success: Published eade7d2d48419acaec1228343071922ec168779f7a6dd46405c9605c5652b43e:current_account-current_region
aws-cdk-rds-proxy: creating CloudFormation changeset...
aws-cdk-rds-proxy |  0/17 | 9:21:23 | UPDATE_IN_PROGRESS   | AWS::CloudFormation::Stack                  | aws-cdk-rds-proxy User Initiated
aws-cdk-rds-proxy |  0/17 | 9:21:29 | CREATE_IN_PROGRESS   | AWS::EC2::SecurityGroup                     | Proxy/ProxySecurityGroup (ProxyProxySecurityGroupC42FC3CE)
aws-cdk-rds-proxy |  0/17 | 9:21:29 | CREATE_IN_PROGRESS   | AWS::IAM::Role                              | Proxy/IAMRole (ProxyIAMRole2FE8AB0F)
aws-cdk-rds-proxy |  0/17 | 9:21:29 | CREATE_IN_PROGRESS   | AWS::RDS::DBSubnetGroup                     | dbCluster/Subnets/Default (dbClusterSubnets03B9B0E1)
aws-cdk-rds-proxy |  0/17 | 9:21:29 | CREATE_IN_PROGRESS   | AWS::EC2::SecurityGroup                     | dbCluster/SecurityGroup (dbClusterSecurityGroupCAA1A91F)
aws-cdk-rds-proxy |  0/17 | 9:21:30 | CREATE_IN_PROGRESS   | AWS::SecretsManager::Secret                 | dbCluster/Secret (dbClusterSecretCEA6D7B6)
aws-cdk-rds-proxy |  0/17 | 9:21:30 | CREATE_IN_PROGRESS   | AWS::IAM::Role                              | Proxy/IAMRole (ProxyIAMRole2FE8AB0F) Resource creation Initiated
aws-cdk-rds-proxy |  0/17 | 9:21:30 | UPDATE_FAILED        | AWS::RDS::DBProxy                           | dbProxy (dbProxy3B89EAF2) Replacement type updates not supported on stack with disable-rollback.
aws-cdk-rds-proxy |  0/17 | 9:21:31 | CREATE_IN_PROGRESS   | AWS::SecretsManager::Secret                 | dbCluster/Secret (dbClusterSecretCEA6D7B6) Resource creation Initiated
aws-cdk-rds-proxy |  1/17 | 9:21:31 | CREATE_COMPLETE      | AWS::SecretsManager::Secret                 | dbCluster/Secret (dbClusterSecretCEA6D7B6)
aws-cdk-rds-proxy |  1/17 | 9:21:32 | CREATE_IN_PROGRESS   | AWS::RDS::DBSubnetGroup                     | dbCluster/Subnets/Default (dbClusterSubnets03B9B0E1) Resource creation Initiated
aws-cdk-rds-proxy |  2/17 | 9:21:32 | CREATE_COMPLETE      | AWS::RDS::DBSubnetGroup                     | dbCluster/Subnets/Default (dbClusterSubnets03B9B0E1)
aws-cdk-rds-proxy |  2/17 | 9:21:34 | CREATE_IN_PROGRESS   | AWS::EC2::SecurityGroup                     | dbCluster/SecurityGroup (dbClusterSecurityGroupCAA1A91F) Resource creation Initiated
aws-cdk-rds-proxy |  2/17 | 9:21:34 | CREATE_IN_PROGRESS   | AWS::EC2::SecurityGroup                     | Proxy/ProxySecurityGroup (ProxyProxySecurityGroupC42FC3CE) Resource creation Initiated
aws-cdk-rds-proxy |  3/17 | 9:21:36 | CREATE_COMPLETE      | AWS::EC2::SecurityGroup                     | Proxy/ProxySecurityGroup (ProxyProxySecurityGroupC42FC3CE)
aws-cdk-rds-proxy |  4/17 | 9:21:36 | CREATE_COMPLETE      | AWS::EC2::SecurityGroup                     | dbCluster/SecurityGroup (dbClusterSecurityGroupCAA1A91F)
aws-cdk-rds-proxy |  4/17 | 9:21:38 | CREATE_IN_PROGRESS   | AWS::RDS::DBCluster                         | dbCluster (dbClusterE86E47AE)
aws-cdk-rds-proxy |  4/17 | 9:21:41 | CREATE_IN_PROGRESS   | AWS::RDS::DBCluster                         | dbCluster (dbClusterE86E47AE) Resource creation Initiated
aws-cdk-rds-proxy |  5/17 | 9:21:49 | CREATE_COMPLETE      | AWS::IAM::Role                              | Proxy/IAMRole (ProxyIAMRole2FE8AB0F)
 5/17 Currently in progress: aws-cdk-rds-proxy, dbClusterE86E47AE
aws-cdk-rds-proxy |  6/17 | 9:22:43 | CREATE_COMPLETE      | AWS::RDS::DBCluster                         | dbCluster (dbClusterE86E47AE)
aws-cdk-rds-proxy |  6/17 | 9:22:46 | CREATE_IN_PROGRESS   | AWS::RDS::DBInstance                        | dbCluster/Instance1 (dbClusterInstance1BCE092AC)
aws-cdk-rds-proxy |  6/17 | 9:22:46 | CREATE_IN_PROGRESS   | AWS::RDS::DBInstance                        | dbCluster/Instance2 (dbClusterInstance20BA1ECD9)
aws-cdk-rds-proxy |  6/17 | 9:22:46 | CREATE_IN_PROGRESS   | AWS::EC2::SecurityGroupIngress              | dbCluster/SecurityGroup/from awscdkrdsproxyProxyProxySecurityGroup9F179E6F:{IndirectPort} (dbClusterSecurityGroupfromawscdkrdsproxyProxyProxySecurityGroup9F179E6FIndirectPortED421002)
aws-cdk-rds-proxy |  6/17 | 9:22:46 | CREATE_IN_PROGRESS   | AWS::EC2::SecurityGroupIngress              | dbCluster/SecurityGroup/from awscdkrdsproxyProxyProxySecurityGroup9F179E6F:{IndirectPort} (dbClusterSecurityGroupfromawscdkrdsproxyProxyProxySecurityGroup9F179E6FIndirectPortED421002) Resource creation Initiated
aws-cdk-rds-proxy |  6/17 | 9:22:47 | CREATE_IN_PROGRESS   | AWS::SecretsManager::SecretTargetAttachment | dbCluster/Secret/Attachment (dbClusterSecretAttachmentAB67A752)
aws-cdk-rds-proxy |  7/17 | 9:22:47 | CREATE_COMPLETE      | AWS::EC2::SecurityGroupIngress              | dbCluster/SecurityGroup/from awscdkrdsproxyProxyProxySecurityGroup9F179E6F:{IndirectPort} (dbClusterSecurityGroupfromawscdkrdsproxyProxyProxySecurityGroup9F179E6FIndirectPortED421002)
aws-cdk-rds-proxy |  7/17 | 9:22:49 | CREATE_IN_PROGRESS   | AWS::RDS::DBInstance                        | dbCluster/Instance1 (dbClusterInstance1BCE092AC) Resource creation Initiated
aws-cdk-rds-proxy |  7/17 | 9:22:49 | CREATE_IN_PROGRESS   | AWS::SecretsManager::SecretTargetAttachment | dbCluster/Secret/Attachment (dbClusterSecretAttachmentAB67A752) Resource creation Initiated
aws-cdk-rds-proxy |  8/17 | 9:22:49 | CREATE_COMPLETE      | AWS::SecretsManager::SecretTargetAttachment | dbCluster/Secret/Attachment (dbClusterSecretAttachmentAB67A752)
aws-cdk-rds-proxy |  8/17 | 9:22:49 | CREATE_IN_PROGRESS   | AWS::RDS::DBInstance                        | dbCluster/Instance2 (dbClusterInstance20BA1ECD9) Resource creation Initiated
aws-cdk-rds-proxy |  8/17 | 9:22:51 | CREATE_IN_PROGRESS   | AWS::RDS::DBProxy                           | Proxy (ProxyCB0DFB71)
aws-cdk-rds-proxy |  8/17 | 9:22:51 | CREATE_IN_PROGRESS   | AWS::IAM::Policy                            | Proxy/IAMRole/DefaultPolicy (ProxyIAMRoleDefaultPolicy59EB0117)
aws-cdk-rds-proxy |  8/17 | 9:22:52 | CREATE_IN_PROGRESS   | AWS::IAM::Policy                            | Proxy/IAMRole/DefaultPolicy (ProxyIAMRoleDefaultPolicy59EB0117) Resource creation Initiated
aws-cdk-rds-proxy |  8/17 | 9:22:53 | CREATE_IN_PROGRESS   | AWS::RDS::DBProxy                           | Proxy (ProxyCB0DFB71) Resource creation Initiated
aws-cdk-rds-proxy |  9/17 | 9:23:09 | CREATE_COMPLETE      | AWS::IAM::Policy                            | Proxy/IAMRole/DefaultPolicy (ProxyIAMRoleDefaultPolicy59EB0117)
 9/17 Currently in progress: aws-cdk-rds-proxy, dbClusterInstance1BCE092AC, dbClusterInstance20BA1ECD9, ProxyCB0DFB71
aws-cdk-rds-proxy | 10/17 | 9:27:08 | CREATE_COMPLETE      | AWS::RDS::DBProxy                           | Proxy (ProxyCB0DFB71)
10/17 Currently in progress: aws-cdk-rds-proxy, dbClusterInstance1BCE092AC, dbClusterInstance20BA1ECD9
aws-cdk-rds-proxy | 11/17 | 9:30:15 | CREATE_COMPLETE      | AWS::RDS::DBInstance                        | dbCluster/Instance1 (dbClusterInstance1BCE092AC)
11/17 Currently in progress: aws-cdk-rds-proxy, dbClusterInstance20BA1ECD9
aws-cdk-rds-proxy | 12/17 | 9:35:58 | CREATE_COMPLETE      | AWS::RDS::DBInstance                        | dbCluster/Instance2 (dbClusterInstance20BA1ECD9)
aws-cdk-rds-proxy | 12/17 | 9:36:05 | CREATE_IN_PROGRESS   | AWS::RDS::DBProxyTargetGroup                | Proxy/ProxyTargetGroup (ProxyProxyTargetGroupB462B5C5)
aws-cdk-rds-proxy | 12/17 | 9:36:07 | CREATE_IN_PROGRESS   | AWS::RDS::DBProxyTargetGroup                | Proxy/ProxyTargetGroup (ProxyProxyTargetGroupB462B5C5) Resource creation Initiated
12/17 Currently in progress: aws-cdk-rds-proxy, ProxyProxyTargetGroupB462B5C5
aws-cdk-rds-proxy | 13/17 | 9:41:50 | CREATE_COMPLETE      | AWS::RDS::DBProxyTargetGroup                | Proxy/ProxyTargetGroup (ProxyProxyTargetGroupB462B5C5)
aws-cdk-rds-proxy | 13/17 | 9:41:51 | UPDATE_IN_PROGRESS   | AWS::CloudFormation::Stack                  | aws-cdk-rds-proxy Replacement type updates not supported on stack with disable-rollback.
aws-cdk-rds-proxy | 14/17 | 9:41:55 | UPDATE_ROLLBACK_COMP | AWS::RDS::DBProxy                           | dbProxy (dbProxy3B89EAF2) Rollback succeeded for the failed resources.
aws-cdk-rds-proxy | 14/17 | 9:41:55 | UPDATE_FAILED        | AWS::CloudFormation::Stack                  | aws-cdk-rds-proxy The following resource(s) failed to update: [dbProxy3B89EAF2].

Failed resources:
aws-cdk-rds-proxy | 9:21:30 | UPDATE_FAILED        | AWS::RDS::DBProxy                           | dbProxy (dbProxy3B89EAF2) Replacement type updates not supported on stack with disable-rollback.

 ❌  aws-cdk-rds-proxy failed: Error: The stack named aws-cdk-rds-proxy failed to deploy: UPDATE_FAILED (The following resource(s) failed to update: [dbProxy3B89EAF2]. ): Replacement type updates not supported on stack with disable-rollback.
    at FullCloudFormationDeployment.monitorDeployment (/Users/yhirokiy/works/ymhiroki/aws-cdk/packages/aws-cdk/lib/api/deploy-stack.ts:505:13)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at deployStack (/Users/yhirokiy/works/ymhiroki/aws-cdk/packages/aws-cdk/lib/cdk-toolkit.ts:265:24)
    at /Users/yhirokiy/works/ymhiroki/aws-cdk/packages/aws-cdk/lib/deploy.ts:39:11
    at run (/Users/yhirokiy/works/ymhiroki/aws-cdk/node_modules/p-queue/dist/index.js:163:29)

 ❌ Deployment failed: Error: Stack Deployments Failed: Error: The stack named aws-cdk-rds-proxy failed to deploy: UPDATE_FAILED (The following resource(s) failed to update: [dbProxy3B89EAF2]. ): Replacement type updates not supported on stack with disable-rollback.
    at Object.exports.deployStacks (/Users/yhirokiy/works/ymhiroki/aws-cdk/packages/aws-cdk/lib/deploy.ts:61:11)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at CdkToolkit.deploy (/Users/yhirokiy/works/ymhiroki/aws-cdk/packages/aws-cdk/lib/cdk-toolkit.ts:339:7)
    at initCommandLine (/Users/yhirokiy/works/ymhiroki/aws-cdk/packages/aws-cdk/lib/cli.ts:374:12)

Stack Deployments Failed: Error: The stack named aws-cdk-rds-proxy failed to deploy: UPDATE_FAILED (The following resource(s) failed to update: [dbProxy3B89EAF2]. ): Replacement type updates not supported on stack with disable-rollback.
  FAILED     integ.proxy-integ.proxy (undefined/us-east-1) 4202.281s
      Integration test failed: Error: Command exited with status 1

Test Results:

Tests:    1 failed, 1 total
Error: Some integration tests failed!
    at main (/Users/yhirokiy/works/ymhiroki/aws-cdk/packages/@aws-cdk/integ-runner/lib/cli.js:147:23)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Fixes must contain a change to an integration test file and the resulting snapshot.
	❌ The title of this pull request does not follow the Conventional Commits format, see https://www.conventionalcommits.org/.

PRs must pass status checks before we can provide a meaningful review.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: ce422a7da2928c258413a865f7426754215296b8
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e6c9d9e8-2ab4-4b94-9fba-b4de81fe4d7e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Fixes must contain a change to an integration test file and the resulting snapshot.

PRs must pass status checks before we can provide a meaningful review.

TheRealAmazonKendra reviewed (dismissed):
Thank you for your contribution! Just a couple comments on this:

1. You'll need to update the integ test to use our `IntegTest` construct and then add 
```
cdkCommandOptions: {
    deploy: {
      args: {
        rollback: true,
      },
    },
  },
```
After the test case. See [this test](https://github.com/aws/aws-cdk/blob/6c606d0f103681db2632d614843233632327693e/packages/%40aws-cdk/lambda-layer-kubectl/test/integ.kubectl-layer.ts#L42) for a good example.

2. The title of a fix should describe the bug, not the fix you're implementing. Please update the title.

Once we have a working build, we'll provide a more meaningful review.

aws-cdk-automation commented:
This PR cannot be merged because it has conflicts. Please resolve them. The PR will be considered stale and closed if it remains in an unmergeable state.

aws-cdk-automation commented:
This PR cannot be merged because it has conflicts. Please resolve them. The PR will be considered stale and closed if it remains in an unmergeable state.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: ec474337709aa9538f41e35a457975388ad42d51
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6165214c-f08c-41a3-805f-545fd6b83a7d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: d0d4c5cbd7247e9c652d69b10c1a2b8d3c56402d
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8dda58fe-959f-49e1-a741-67b196a105df%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 5f55fc36dfec93ac1004b5a63c09c1d5b00c7549
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=359a4dba-b467-443e-845b-1dd93ab5bca4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

ymhiroki commented:
Thanks for nice advice, @TheRealAmazonKendra !

I added snapshots for integration tests and passed the CIs. I beg your reviews.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 0a15a3088be09d40a0d8f9556422d15d8730caa2
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=89f4720a-06d9-4883-b157-30fc51a64c5b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
